hpOneView/resources/networking/interconnect_types.py

```diff
@@ -0,0 +1,99 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+###
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from future import standard_library
+
+standard_library.install_aliases()
+
+__title__ = 'Interconnect Types'
+__version__ = '0.0.1'
+__copyright__ = '(C) Copyright (2012-2016) Hewlett Packard Enterprise ' \
+                ' Development LP'
+__license__ = 'MIT'
+__status__ = 'Development'
+
+from hpOneView.resources.resource import ResourceClient
+
+
+class InterconnectTypes(object):
+    URI = '/rest/interconnect-types'
+
+    def __init__(self, con):
+        self._connection = con
+        self._client = ResourceClient(con, self.URI)
+
+    def get_all(self, start=0, count=-1, filter='', sort=''):
+        """
+        Gets a paginated collection of all interconnect types based on the specified parameters.
+        Filters can be used in the URL to control the number of interconnect types that are returned.
+        With no filters specified, the API returns all interconnect types.
+
+        Args:
+            start:
+                The first item to return, using 0-based indexing.
+                If not specified, the default is 0 - start with the first available item.
+            count:
+                The number of resources to return. A count of -1 requests all the items.
+                The actual number of items in the response may differ from the requested
+                count if the sum of start and count exceed the total number of items, or
+                if returning the requested number of items would take too long.
+            filter:
+                A general filter/query string to narrow the list of items returned. The
+                default is no filter - all resources are returned.
+            sort:
+                The sort order of the returned data set. By default, the sort order is based
+                on create time, with the oldest entry first.
+
+        Returns: dict
+
+        """
+        return self._client.get_all(start, count, filter=filter, sort=sort)
+
+    def get(self, id):
```

Comment:
I suggest accept either the ID or the URI of the resource in the methods like this, that  need a resource reference. 
(E.g.:  'e28949fa-8681-433d-9808-a9146feb8048' or '/rest/interconnect-types/28949fa-8681-433d-9808-a9146feb8048')
We started use this recently (look to networking/logical_interconnect_groups.py), after notice that the created resouces do not have an attribute 'id', however they have an attribute 'uri' that is the reference to the resource.